### PR TITLE
Search & Previous-Next features in homepage

### PR DESCRIPTION
**Pull request by ADITIRAI290698**

### CODE REVIEW

File: dj/movies/views.py [hunk 1]
```diff
@@ -47,11 +46,11 @@ def list_movie(req: django.http.HttpRequest):
...
-    p = Paginator(Movie.objects.all(), items)
+    movie_list = Movie.objects.all()[(p_number - 1) * items: p_number * items]
+    # movies = Movie.objects.all()
+    # p = Paginator(Movie.objects.all(), items)
 
-    movie_list = p.get_page(page_number)
+    # movie_list = p.get_page(page_number)
     list_of_top_movies = []
     list_of_bottom_movies = []
     index = 0
```
Comment (AbrarNitk) [inline dj/movies/views.py]: Remove unused code and comments

File: dj/movies/views.py [hunk 2]
```diff
@@ -95,23 +94,57 @@ def search_movie(req: django.http.HttpRequest):
...
+@csrf_exempt
+def search_page(req: django.http.HttpRequest):
+
+
+    target_movie_name = req.GET.get("search_movie", "default")
+
+    search_for = req.GET.get("movie", None)
+    # print(f"searching for {search_for}")
+
+    searched_movie_list = Movie.objects.filter(title__startswith=search_for).order_by("-updated_on")
```
Comment (AbrarNitk) [inline dj/movies/views.py]: You used three names with the same purpose. `target_movie_name`, `search_movie`, `search_for`.

File: dj/movies/views.py [hunk 3]
```diff
@@ -95,23 +94,57 @@ def search_movie(req: django.http.HttpRequest):
...
+
+
+    list_of_searched_movies = []
+    for movie in searched_movie_list:
+        rating = give_rating(movie.id)
+        item = {
+            "title": movie.title,
+            "url": "http://127.0.0.1:8000/movie/?id=" + str(movie.id),
```
Comment (AbrarNitk) [inline dj/movies/views.py]: Why do we need to pass the host address, `/movie/?id=` should work.

**Review by AbrarNitk — CHANGES_REQUESTED** [review]

File: dj/movies/views.py (hunk 2 above)
Comment (AviralVerma13) [reply dj/movies/views.py]: using search_for only now.

File: dj/movies/views.py (hunk 3 above)
Comment (AviralVerma13) [reply dj/movies/views.py]: yes it's working.

**Comment by AviralVerma13** [comment]
contents moved to [correct movie slicing](https://github.com/FifthTry/movie-db/pull/33)

**Pull request closed without merging**